catalog/transformer/catalog-transformer-tika-input/src/main/java/ddf/catalog/transformer/input/tika/TikaInputTransformer.java

```diff
@@ -121,22 +160,57 @@ public Metacard transform(InputStream input, String id)
             }
 
             Parser parser = new AutoDetectParser();
-            ToXMLContentHandler handler = new ToXMLContentHandler();
+            ToXMLContentHandler xmlContentHandler = new ToXMLContentHandler();
+            ToTextContentHandler textContentHandler = null;
+            ContentHandler contentHandler;
+            if (!contentMetadataExtractors.isEmpty()) {
+                textContentHandler = new ToTextContentHandler();
+                contentHandler = new TeeContentHandler(xmlContentHandler, textContentHandler);
+            } else {
+                contentHandler = xmlContentHandler;
+            }
+
             TikaMetadataExtractor tikaMetadataExtractor = new TikaMetadataExtractor(parser,
-                    handler);
+                    contentHandler);
 
             Metadata metadata;
             try (InputStream inputStreamCopy = fileBackedOutputStream.asByteSource()
                     .openStream()) {
                 metadata = tikaMetadataExtractor.parseMetadata(inputStreamCopy, new ParseContext());
             }
 
-            String metadataText = handler.toString();
+            String metadataText = xmlContentHandler.toString();
             if (templates != null) {
                 metadataText = transformToXml(metadataText);
             }
 
-            Metacard metacard = MetacardCreator.createBasicMetacard(metadata, id, metadataText);
+            Metacard metacard;
+            if (textContentHandler != null) {
+                String plainText = textContentHandler.toString();
+
+                Set<AttributeDescriptor> attributes = contentMetadataExtractors.values()
```

Comment:
Noticing a fair amount of code duplication between these two transformers. Any chance some of that could be refactored away?